x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/InferenceResultFieldMapper.java

```diff
@@ -364,9 +370,13 @@ public static void applyFieldInference(
                 results.getWriteableName()
             );
         }
-        Map<String, Object> fieldMap = new LinkedHashMap<>();
+
+        Map<String, Object> fieldMap = (Map<String, Object>) inferenceMap.computeIfAbsent(field, s -> new LinkedHashMap<>());
         fieldMap.putAll(new SemanticTextModelSettings(model).asMap());
```

Comment:
We could be potentially adding the model settings to the field map multiple times. Any concerns with this?

Reply:
It's possible - they would overwrite existing settings. Besides minimal performance improvements I don't see a problem with this